PLUGINS.md

```diff
@@ -0,0 +1,69 @@
+# Writing and Using IOpipe Plugins for Java
+
+This document contains information on how to write and use plugins for IOpipe.
+
+For every invocation that is performed there is a unique object which
+represents the state of the plugin for each plugin as they are needed. This
+execution state is initialized when the plugin is first referenced. When the
+method finishes execution, the state is removed and nothing more is done with
+it. This mechanism can be used to keep track of which custom metrics should
+be written to the report for example. The execution state is an implementation
+of the class `com.iopipe.plugin.IOpipePluginExecution`.
+
+# Using Plugins
+
+All plugins are referred to by their execution state, which is an
+implementation of the class `com.iopipe.plugin.IOpipePluginExecution`.
+To obtain the execution state of a plugin
```

Comment:
This appears disjointed - is something missing?

Reply:
I removed the truncated sentence.